docs/painless/painless-basic-syntax.asciidoc

```diff
@@ -0,0 +1,304 @@
+
+*Examples:*
+
+Single-line and multi-line comments where `<code>` represents tokens within a
+script.
+
+[source,Java]
+----
+// single-line comment
+
+<code> // single-line comment
+
+/* multi-
+   line
+   comment */
+
+<code> /* multi-line
+          comment */ <code>
+
+<code> /* multi-line comment */ <code>
+----
+
+[[keywords]]
+==== Keywords
+
+The keywords in the table below are reserved for built-in language
+features. These keywords cannot be used as identifiers or types.
+
+[cols="^1,^1,^1,^1,^1"]
+|====
+| if | else | while | do | for
+| in | continue | break | return | new
+| try | catch | throw | this | instanceof
+|====
+
+[[literals]]
+==== Literals
+
+Use literals to specify different types of values directly in a script.
+
+[[integers]]
+===== Integers
+
+Use integer literals to specify an integer value in decimal, octal, or hex
+notation of the <<primitive-types, primitive types>> int, long, float, or
+double. Use the following single letter designations to specify the
+<<primitive-types, primitive type>>: `l` or `L` for `long`, `f` or `F` for
+`float`, and `d` or `D` for `double`. If not specified, the type defaults to
+`int`.  Use `0` as a prefix to specify an integer literal as octal, and use
+`0x` or `0X` as a prefix to specify an integer literal as hex.
+
+*Grammar:*
+[source,ANTLR4]
+----
+INTEGER: '-'? ( '0' | [1-9] [0-9]* ) [lLfFdD]?;
+OCTAL:   '-'? '0' [0-7]+ [lL]?;
+HEX:     '-'? '0' [xX] [0-9a-fA-F]+ [lL]?;
+----
+
+*Examples:*
+
+Integer literals of `int 0`, `double 0.0`, `long 1234`,
+`float -90.0`, `int -18` in octal, and `int 3882` in hex.
+
+[source,Java]
+----
+0
+0D
+1234L
+-90f
+-022
+0xF2A
+----
+
+[[floats]]
+===== Floats
+
+Use floating point literals to specify a floating point value of the
+<<primitive-types, primitive types>> float or double. Use the following single
+letter designations for the <<primitive-types, primitive type>>: `f` for
+`float` and `d` for `double`. If not specified, the type defaults to `double`.
+
+*Grammar:*
+[source,ANTLR4]
+----
+DECIMAL: '-'? ( '0' | [1-9] [0-9]* ) (DOT [0-9]+)? EXPONENT? [fFdD]?;
+EXPONENT: ( [eE] [+\-]? [0-9]+ );
+----
+
+*Examples:*
+
+Floating point literals of `double 0.0`, `double 1000000.0` in
+exponent notation, `double 0.977777`, `double -126.34`, and `float 89.9`.
+
+[source,Java]
+----
+0.0
+1E6
+0.977777
+-126.34
+89.9F
+----
+
+[[strings]]
+===== Strings
+
+Use string literals to specify string values of the
+<<string-type, String type>> with either single or double quotes.
+Use a `\"` token to include a double-quote as part of a double-quoted string
+literal. Use a `\'` token to include a single-quote as part of a single-quoted
+string literal.  Use a `\\` token to include a backslash as part of any string
+literal.
+
+*Grammar:*
+[source,ANTLR4]
+----
+STRING: ( '"'  ( '\\"'  | '\\\\' | ~[\\"] )*? '"'  )
+      | ( '\'' ( '\\\'' | '\\\\' | ~[\\'] )*? '\'' );
+----
+
+*Examples:*
+
+String literals using both single-quotes and double-quotes.
+
```

Comment:
I'd probably split this into "Single quoted string literals" and "Double quoted string literals" examples.

Reply:
Done.